src/test/java/org/jenkinsci/plugins/oic/OicSecurityRealmFipsTest.java

```diff
@@ -0,0 +1,98 @@
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+public class OicSecurityRealmFipsTest {
+
+    @ClassRule
+    public static FlagRule<String> fipsFlag = FlagRule.systemProperty(FIPS140.class.getName() + ".COMPLIANCE", "true");
+
+    @Rule
+    public NonFailingOnStartupJenkinsRule j = new NonFailingOnStartupJenkinsRule();
+
+    @Test
+    @WithoutJenkins
+    public void settingNonCompliantValuesNotAllowedTest() throws IOException {
+        OicSecurityRealm realm = new OicSecurityRealm("clientId", Secret.fromString("secret"), null, false);
+        IllegalArgumentException ex = assertThrows(
+                IllegalArgumentException.class,
+                () -> new OicSecurityRealm("clientId", Secret.fromString("secret"), null, true));
+        assertThat(
+                "Exception contains the reason",
+                ex.getLocalizedMessage(),
+                containsString("SSL verification can not be disabled"));
+        realm.setDisableTokenVerification(false);
+        ex = assertThrows(IllegalArgumentException.class, () -> realm.setDisableTokenVerification(true));
+        assertThat(
+                "Exception contains the reason",
+                ex.getLocalizedMessage(),
+                containsString("Token verification can not be disabled"));
+    }
+
+    @Test
+    @WithoutJenkins
+    public void validationWarnsOfInvalidValuesTest() {
+        OicSecurityRealm.DescriptorImpl descriptor = new OicSecurityRealm.DescriptorImpl();
+        FormValidation response = descriptor.doCheckDisableSslVerification(true);
+        assertThat("Shows an error", response.kind, is(FormValidation.Kind.ERROR));
+        assertThat(
+                "States SSL verification can not be disabled",
+                response.getMessage(),
+                containsString("SSL verification can not be disabled"));
+        response = descriptor.doCheckDisableSslVerification(false);
+        assertThat("Validation is ok", response.kind, is(FormValidation.Kind.OK));
+
+        response = descriptor.doCheckDisableTokenVerification(true);
+        assertThat("Shows an error", response.kind, is(FormValidation.Kind.ERROR));
+        assertThat(
+                "States token verification can not be disabled",
+                response.getMessage(),
+                containsString("Token verification can not be disabled"));
+        response = descriptor.doCheckDisableTokenVerification(false);
+        assertThat("Validation is ok", response.kind, is(FormValidation.Kind.OK));
+    }
+
+    // This test is not strictly needed as per
+    // https://github.com/jenkinsci/jep/blob/master/jep/237/README.adoc#backwards-compatibility
+    // Just adding it to cover corner cases
+    @Test
+    @LocalData
+    public void failsOnMigrationTest() {
+        assertTrue("We should get a ReactorException, startup failed", j.getError() instanceof ReactorException);
```

Comment:
I'm not fan of ussing `assertTrue` or `assertFalse` with non boolean fields as you miss information. If you use `assertThat` with https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/IsInstanceOf.html you will know the actual value that is not matching the expected one